src/components/EnrolmentOfficerFormPanel.js

```diff
@@ -42,19 +41,36 @@ const EnrolmentOfficerFormPanel = (props) => {
         }
       }
     `,
-      { system_id: 1 }, // EO System Role is 1
-    );
-    const isValid = !isLoading;
-    useEffect(() => {
-      if (isValid & isEnabled & !hasRole) {
-        const role = data.role.edges[0].node;
-        const roles = !!edited.roles ? edited.roles : [];
-        roles.push(role);
-        edited.roles = roles;
-        onEditedChanged(edited);
+    { system_id: 1 }, // EO System Role is 1
+  );
+  const isValid = !isLoading;
+
+  useEffect(() => {
+    const roles = edited.roles ? edited.roles : [];
```

Comment:
Can you try with optional chaining? 
I think we can handle this by
`const roles = edited?.roles ?? [];`
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

Reply:
Changed.